src/OpenLoco/IndustryManager.cpp

```diff
@@ -190,18 +190,238 @@ namespace OpenLoco::IndustryManager
+            {
+                auto tile = Map::TileManager::get(randomPos);
+                auto* surface = tile.surface();
+                auto* climateObj = ObjectManager::get<ClimateObject>();
+                if (surface == nullptr || surface->baseZ() < climateObj->summerSnowLine)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtBelowSnowLine)
+            {
+                auto tile = Map::TileManager::get(randomPos);
+                auto* surface = tile.surface();
+                auto* climateObj = ObjectManager::get<ClimateObject>();
+                if (surface == nullptr || surface->baseZ() > climateObj->winterSnowLine)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtOnFlatGround)
+            {
+                if (mountainHeight(randomPos) > 32)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtInDesert)
+            {
+                if (Map::TileManager::countSurroundingDesertTiles(randomPos) < 100)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtNearDesert)
+            {
+                if (Map::TileManager::countSurroundingDesertTiles(randomPos) >= 70)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtNearWater)
+            {
+                if (Map::TileManager::countSurroundingWaterTiles(randomPos) < 10)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtAwayFromWater)
+            {
+                if (Map::TileManager::countSurroundingWaterTiles(randomPos) != 0)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtOnWater)
+            {
+                auto tile = Map::TileManager::get(randomPos);
+                auto* surface = tile.surface();
+                if (surface != nullptr && surface->water() == 0)
+                {
+                    continue;
+                }
+            }
+            if (indObj->flags & IndustryObjectFlags::builtNearTown)
+            {
+                auto res = TownManager::getClosestTownAndUnk(randomPos);
+                if (!res.has_value())
+                {
+                    continue;
+                }
+                const auto& [townId, builtUp] = *res;
+                if (builtUp == 0)
+                {
+                    const auto* town = TownManager::get(townId);
+                    if (Math::Vector::manhattanDistance(randomPos, Map::Pos2{ town->x, town->y }) > 576)
```

Comment:
constants for these numbers? (576, 768, 25, 3, etc)

Reply:
Doesn't seem that necessary to me if they're only used here. Not against it either though.

Reply:
I've added some constants.